### PR TITLE
feat(demo): reorganize components and dialog refactor

### LINKED ISSUE #152: Dialog component
**As a Pillarbox developer**, I'd like the `dialog` element to be a component, to facilitate reuse.

## Acceptance Criteria

- Create a lit component

### LINKED ISSUE #153: Reorganizing css by component
**As a Pillarbox developer**, I'd like my CSS to be organized by component for easier access.

## Acceptance Criteria

- each component has its own `directory`
- each directory contains the component's `js`
- each directory contains the component's `css`

### PR DESCRIPTION
**Pull request by jboix**
## Description

Closes #152
Closes #153

This commit focuses on restructuring and enhancing the codebase for improved organization and maintainability. The key aspects of this commit include:

### 1. Folder Layout Reorganization.

The demo folder now follows a revised layout structure:

- **core:** Contains common modules and components, such as the spinner, content-link, dialog, spa router, etc.
- **layout:** Houses pages and major layout elements, including the header, example page, search page, etc.

This restructuring aims to make the demo page more intuitive and modular.

### 2. Refactor `player-dialog` to a web component

The `player-dialog` has been refactored into a web component. This refactor has led to a cleaner separation of concerns, isolating the dialog's styles from the main page.

### 3. Player Query Params Routing Logic

The logic responsible for handling the routing of player query parameters has been relocated to  the `player.js` file. This ensures that the routing-related functionality is centralized within the appropriate file.

### CODE REVIEW

File: demo/src/layout/content/examples/load-media-form-component.js [hunk 1]
```diff
@@ -119,13 +119,13 @@ export class LoadMediaFormComponent extends LitElement {
...
     if (_changedProperties.has('drmSettingsShown') && this.drmSettingsShown) {
-      this.shadowRoot.querySelector('.drm-settings-container').add('active');
+      this.shadowRoot.querySelector('.drm-settings-container').classList.add('active');
     }
   }
 
   #onFormAnimationEnd(e) {
     if (e.target.classList.contains('shrink')) {
-      e.target.classList.add('active');
+      e.target.classList.classList.remove('active');
```
Comment (amtins) [inline demo/src/layout/content/examples/load-media-form-component.js:128]: An extra `classList` for free?

**Review by amtins — CHANGES_REQUESTED** [review]
What about something like ?

image

File: demo/src/layout/content/examples/load-media-form-component.js (hunk 1 above)
Comment (jboix) [reply demo/src/layout/content/examples/load-media-form-component.js]: Done !

**Review by amtins — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**